### PR TITLE
Apollo PCBs firmware modernization

### PR DESCRIPTION
**Pull request by Gondolindrim**
## Description

Modernizes the firmwares for the Acheron Apollo PCBs, adding data-driven configuration and a modernized version of dynamically user-configurable indicators in VIA.

The modernization pretty much consists of moving most of configurations into `info.json`. `keyboard.h` were all removed, and `rules.mk` only contain building optimization configuration. `config.h` also only contain the RGB matrix animation enables and some configuration, like number of LEDs and turn off on sleep.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by tzarc** [comment]
Migrations have to go to `develop` due to other work partially modernising boards.
As you can see there are conflicts -- chances are a bunch of things have already been done.

**Comment by tzarc** [comment]
> I see. Should I reopen a new PR then with the changes on top of `develop`?

Nah just fix the conflicts on the current branch.

**Comment by Gondolindrim** [comment]
Forgive me @tzarc , for I have sinned...

**Comment by Gondolindrim** [comment]
I will be fixing the issues for `87h/gamma` soon.

File: keyboards/acheron/apollo/88htsc/info.json [hunk 1]
```diff
@@ -8,123 +8,217 @@
         "pid": "0x8873",
```
Comment (fauxpark) [inline keyboards/acheron/apollo/88htsc/info.json:8]: Please run `qmk format-json` on all of these.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/acheron/apollo/88htsc/info.json (hunk 1 above)
Comment (Gondolindrim) [reply keyboards/acheron/apollo/88htsc/info.json]: Done in 0415331

File: keyboards/acheron/apollo/87h/gamma/info.json [hunk 2]
```diff
@@ -1,122 +1,226 @@
 {
```
Comment (fauxpark) [inline keyboards/acheron/apollo/87h/gamma/info.json:1]: Same here.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/acheron/apollo/87h/gamma/info.json (hunk 2 above)
Comment (Gondolindrim) [reply keyboards/acheron/apollo/87h/gamma/info.json:1]: Done in [c372580](https://github.com/qmk/qmk_firmware/pull/20539/commits/c372580a521b857fafe6c2a83eea4eddeecba55e).

**Pull request closed without merging**